pkg/pool-weighted/contracts/managed/AaveWeightedPoolFactory.sol

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-pool-utils/contracts/factories/BasePoolFactory.sol";
+
+import "@balancer-labs/v2-vault/contracts/interfaces/IVault.sol";
+import "@balancer-labs/v2-vault/contracts/interfaces/IBasePool.sol";
+
+import "@balancer-labs/v2-asset-manager-utils/contracts/AaveATokenAssetManager.sol";
+
+import "../IWeightedPoolFactory.sol";
+
+contract AaveWeightedPoolFactory is BasePoolFactory {
+    IWeightedPoolFactory public immutable poolFactory;
+    ILendingPool public immutable lendingPool;
+
+    constructor(IWeightedPoolFactory baseFactory, ILendingPool aaveLendingPool)
+        BasePoolFactory(baseFactory.getVault())
+    {
+        poolFactory = baseFactory;
+        lendingPool = aaveLendingPool;
+    }
+
+    /**
+     * @dev Deploys a new `WeightedPool` which invests specified tokens into Aave.
+     */
+    function create(
+        string memory name,
+        string memory symbol,
+        IERC20[] memory tokens,
+        uint256[] memory weights,
+        uint256[] memory managedTokens,
```

Comment:
I found usage if this array slightly unintuitive. What if we simply had an array with the same length as `tokens`, holding `bool managed`? We'd do a few extra iterations, but that seems fine.